base/rational.jl

```diff
@@ -14,8 +14,17 @@ struct Rational{T<:Integer} <: Real
         num2, den2 = (sign(den) < 0) ? divgcd(-num, -den) : divgcd(num, den)
         new(num2, den2)
     end
+
+    function Rational{T}(num::Integer, den::Integer) where T<:BitSigned
+        num == den == zero(T) && __throw_rational_argerror(T)
+        num < 0 && den < 0 && (num === typemin(T) || den === typemin(T)) && __throw_rational_ovferror(num, den)
```

Comment:
This check may have to move after the `divgcd` to handle the `typemin(Int)//-2` case

Reply:
You are right about that.  I made the change.  I see no distinction between `Int < 0` and `sign(Int) < 0` in generated code -- is there a reason to prefer one over the other?

Reply:
not that I can think of?

Reply:
this is an additional gotcha
```
julia> divgcd(-2,typemin(Int64))
(-1, -4611686018427387904)

julia> Rational{Int}(-2,typemin(Int))
Rational{Int64}(1, -4611686018427387904) # both should be positive
```